### DISCUSSION TITLE
Tree structure with checkbox

### DISCUSSION THREAD

**Discussion by mohammad-reza-rahbar** [mantinedev/mantine#6496] [Q&A] [answered]
Is there an easy way to add node selection with checkbox? I know there is a `setSelectedState` function, which works when selecting a node and all its descendants because of `children` property, but because there is no `parent` property for a child node, I don't know how can I have nodes which are indeterminate.

**Comment by rtivital** [comment, marked as answer]
Checkbox state selection is planned to be added as a feature to the Tree component in one of the next minor releases.
